src/palette/internal/palette/paletteworkspace.cpp

```diff
@@ -661,6 +661,7 @@ void PaletteWorkspace::setSearching(bool searching)
     m_searching = searching;
 
     mainPalette = nullptr;
+    mainPaletteController = nullptr;
```

Comment:
Ensures a new UserPaletteController gets created with the correct model - I assume Qt looks after the memory management here.  Alternatively we could add a public method to `UserPaletteController` to update the model but this seems safer.